scripts/bootstrap.sh.in

```diff
@@ -182,43 +182,82 @@ orchestrate_bootstrap() {
     # Grains must be set (in `/etc/salt/grains`) *before* invoking `salt-call`,
     # otherwise grains set during execution won't be taken into account
     # properly.
-    ${SALT_CALL} --local --retcode-passthrough state.sls \
+    ${SALT_CALL} --local --state-output=mixed --retcode-passthrough state.sls \
         metalk8s.node.grains \
         saltenv=metalk8s-@@VERSION \
         pillarenv=metalk8s-@@VERSION
 
-    # TODO: Registry IP should not need a pillar override here
-    local -r registry_ip=$(
+    # TODO: FIX registry IP
+    local registry_ip
+    registry_ip=$(
         ${SALT_CALL} --local grains.get metalk8s:control_plane_ip --out txt \
         | awk '/^local\: /{ print $2 }'
     )
 
-    ${SALT_CALL} --local --retcode-passthrough state.orchestrate \
-        metalk8s.orchestrate.bootstrap_without_master \
+    pillar=(
+      "{"
+      "  'repo': {'local_mode': True},"
+      "  'registry_ip': $registry_ip,"
+      "  'metalk8s': {"
+      "    'endpoints': {"
+      "      'salt-master': {'ip': $registry_ip}"
+      "    }"
+      "  }"
+      "}"
+    )
+
+    ${SALT_CALL} --local --state-output=mixed  --retcode-passthrough state.sls \
+        metalk8s.roles.bootstrap.no-master \
         saltenv=metalk8s-@@VERSION \
         pillarenv=metalk8s-@@VERSION \
-        pillar="{'registry_ip': '$registry_ip'}"
+        pillar="${pillar[*]}"
 
     sync_salt_master
 
-    local -r salt_master_call="crictl exec -i $(get_salt_container)"
+    SALT_MASTER_CALL="crictl exec -i $(get_salt_container)"
     local -r bootstrap_id=$(cat /etc/salt/minion_id)
 
     # Wait for Salt minion to respond before proceeding (may not have completed
     # handshake with the master yet)
-    $salt_master_call salt-run metalk8s_saltutil.wait_minions \
+    ${SALT_MASTER_CALL} salt-run metalk8s_saltutil.wait_minions \
         tgt="$bootstrap_id"
 
-    sync_salt_minion "$bootstrap_id"
+    sync_salt_minion
+
+    # Do CA only first to push informations to the mine
+    ${SALT_CALL} --state-output=mixed --retcode-passthrough state.sls \
+        metalk8s.roles.ca \
+        saltenv=metalk8s-@@VERSION \
+        pillarenv=metalk8s-@@VERSION
+
+    pillar=(
```

Comment:
No real need to define this twice, no? Could set it up 'completely' (i.e. the version as below) above from the start?

Reply:
Also, the `read -r -d '' PILLAR <<- END` version is more legible IMHO ;)

Reply:
If we use read like that we can't use indentation with space we must use tabs. So I not really like

Reply:
We need to define it twice because of the "repo local_mode"

Reply:
OK. And bummer :)